library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -67,6 +67,10 @@ void download() {
             long currentBytesDownloaded = getBytesDownloadedFrom(fileBytesDownloadedMap);
             downloadBatchStatus.update(currentBytesDownloaded, totalBatchSizeBytes);
 
+            if (currentBytesDownloaded == totalBatchSizeBytes && totalBatchSizeBytes != ZERO_BYTES) {
```

Comment:
A version of this was originally in the `DownloadBatchStatus` but it never marked the `status` as `DOWNLOADED` at the persistence layer, just in memory 😬